src/utils/MainApi.js

```diff
@@ -0,0 +1,42 @@
+// description to request to  my API
+
+export default class MainApi {
+  constructor({baseUrl, headers}) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "GET",
+      headers: this._headers
+    })
+    .then((res) => res.ok ? res.json() : Promise.reject(`Error! ${res.statusText}`))
```

Comment:
```.then((res) => res.ok ? res.json() : Promise.reject(`Error! ${res.statusText}`))```

This code is repeated in every request, so it's better to make a special method or function for that checking and use it everywhere here and in `NewApi.js` and in `auth.js`